Projects/Feature/Onboarding/Feature/Sources/InputWakeUpAlarm/Views/CTAButton/DefaultCTAButton.swift

```diff
@@ -0,0 +1,201 @@
+//
+//  DefaultCTAButton.swift
+//  FeatureOnboarding
+//
+//  Created by choijunios on 1/6/25.
+//
+
+import UIKit
+
+import FeatureResources
+
+import SnapKit
+
+protocol DefaultCTAButtonListener: AnyObject {
+    
+    func action(_ action: DefaultCTAButton.Action)
+}
+
+class DefaultCTAButton: UIView {
+    
+    // Action
+    enum Action {
+        case buttonIsTapped
+    }
+    
+    
+    // Sub view
+    private let titleLabel: UILabel = .init()
+    private let fadeView: UIView = .init()
+    
+    
+    // Listener
+    weak var listener: DefaultCTAButtonListener?
+    
+
+    // State
+    private var state: State = .active
+    private var isEnabled: Bool { state == .active }
+    
+    
+    // Gesture
+    private let tapRecognizer: UITapGestureRecognizer = .init()
+    
+    
+    //
+    private let baseFont: R.Font = .body1SemiBold
+    
+    
+    override var intrinsicContentSize: CGSize {
+        
+        .init(
+            width: super.intrinsicContentSize.width,
```

Comment:
의도하신게 width는 외부에서 주입해주길 원한거라면 width를 UIView.noIntrinsicMetric를 써주시면 됩니다.

> If a custom view has no intrinsic size for a given dimension, it can use [noIntrinsicMetric](https://developer.apple.com/documentation/uikit/uiview/nointrinsicmetric) for that dimension.